scenarios/throughput_stress.go

```diff
@@ -41,18 +45,42 @@ const (
 	defaultWorkflowIDPrefix                   = "throughputStress"
 )
 
+type tpsStatusUpdate struct {
+	// CompletedIteration is the iteration that has been completed.
+	CompletedIteration int
+	// CompletedWorkflows is the total number of workflows that have been completed so far.
+	CompletedWorkflows uint64
+}
+
 type tpsExecutor struct {
 	workflowCount atomic.Uint64
 }
 
+// Run executes the throughput stress scenario.
+//
+// It executes `throughputStress` workflows in parallel - up to the configured maximum cocurrency limit - and
+// waits for the results. At the end, it verifies that the total number of executed workflows matches Visibility's count.
+//
+// To resume a previous run, set the following options:
+//
+// --option internal-iterations-resume-from=<value>
+// --option workflow-count-resume-from=<value>
+//
+// Note that the caller is responsible for adjusting the scenario's iterations/timeout accordingly.
 func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error {
 	// Parse scenario options
 	internalIterations := info.ScenarioOptionInt(IterFlag, 5)
 	internalIterTimeout := info.ScenarioOptionDuration(IterTimeout, time.Minute)
+	internalIterResumeFrom := info.ScenarioOptionInt(IterResumeFrom, 0)
+	_, resumingFromPreviousRun := info.ScenarioOptions[IterResumeFrom]
 	continueAsNewCount := info.ScenarioOptionInt(CANEventFlag, 120)
 	workflowIDPrefix := cmp.Or(info.ScenarioOptions[WorkflowIDPrefix], defaultWorkflowIDPrefix)
+	workflowCountResume := info.ScenarioOptionInt(WorkflowCountResumeFrom, 0)
 	nexusEndpoint := info.ScenarioOptions[NexusEndpointFlag] // disabled by default
 	skipSleep := info.ScenarioOptionBool(SkipSleepFlag, false)
+	if info.StatusCallback == nil {
+		info.StatusCallback = func(data any) {}
```

Comment:
I considered doing this further up the callstack, but didn't find a good place to set the default.

Reply:
I could move it into `RunConfiguration` (which has a "set defaults" step)?